frameworks/Java/armeria/src/main/java/hello/services/HelloService.java

```diff
@@ -0,0 +1,57 @@
+package hello.services;
+
+import hello.models.Message;
+
+import java.nio.charset.StandardCharsets;
+import java.text.DateFormat;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.core.JsonProcessingException;
+
+import com.linecorp.armeria.common.HttpData;
+import com.linecorp.armeria.common.HttpHeaderNames;
+import com.linecorp.armeria.common.HttpHeaders;
+import com.linecorp.armeria.common.HttpResponse;
+import com.linecorp.armeria.common.HttpStatus;
+import com.linecorp.armeria.common.MediaType;
+import com.linecorp.armeria.server.annotation.Get;
+import com.linecorp.armeria.server.annotation.ProducesJson;
+
+public class HelloService
+{
+  private static final byte[]       PLAINTEXT      = "Hello, World!".getBytes(
+      StandardCharsets.UTF_8);
+  private static final ObjectMapper MAPPER         = new ObjectMapper();
+  private static final DateFormat   DATE_FORMATTER = new SimpleDateFormat(
```

Comment:
This looks unsafe to me.  A single `SimpleDateFormat` instance should not be used concurrently from multiple threads.  Unless this framework somehow guarantees that only one thread will be executing any one of your request handler methods at a given time (which would have to be disastrous for performance, wouldn't it?), this code is incorrect.

In cases where you have to use `SimpleDateFormat`, I suggest constructing a new instance each time you have to format a date.  In other words, replace uses of `DATE_FORMATTER` with `new SimpleDateFormat(...)`.  (You may be able to optimize that by using a `ThreadLocal<SimpleDateFormat>`, but it would make the code slightly more complicated.  In most circumstances I don't bother with `ThreadLocal` for stuff like this.)

However, both `SimpleDateFormat` and `java.util.Date` should be avoided in modern Java code.  Those are considered part of the legacy date-time libraries in Java.  Java 8 introduced the `java.time` packages as a replacement.  In particular, `java.time.format.DateTimeFormatter` is the thread-safe alternative to `SimpleDateFormat`.  It has a built-in constant for the standard HTTP date format.

Try out the code from this StackOverflow answer: https://stackoverflow.com/a/26367834

```java
DateTimeFormatter.RFC_1123_DATE_TIME.format(ZonedDateTime.now(ZoneOffset.UTC))
```